core/src/qrsignatures.c

```diff
@@ -7,40 +7,59 @@
 #include "memzero.h"
 #include "nist256p1.h"
 
+#include <assert.h>
 #include <stdbool.h>
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
 
-static const uint8_t QR_PUBKEY[65] = _QR_PUBKEY;
-bool check_qrsignature(const uint8_t * const data, size_t data_len, const uint8_t * const signature){
-  return check_qrsignature_pub(data, data_len, signature, QR_PUBKEY);
-}
+static const uint8_t QR_PUBKEY[QRSIGNATURE_PUBKEY_LEN] = _QR_PUBKEY;
 
-bool check_qrsignature_pub(const uint8_t * const data, size_t data_len, const uint8_t * const signature, const uint8_t * const pub){
+bool check_qrsignature(
+    const uint8_t* const data,
+    const size_t data_len,
+    const uint8_t* const signature,
+    const size_t signature_len) {
+  static_assert(sizeof(QR_PUBKEY) == QRSIGNATURE_PUBKEY_LEN, "sizeof(QR_PUBKEY) must equal QRSIGNATURE_PUBKEY_LEN");
+  return check_qrsignature_pub(data, data_len, signature, signature_len, QR_PUBKEY, sizeof(QR_PUBKEY));
+}
 
-  if (data_len == 0){
+bool check_qrsignature_pub(
+    const uint8_t* const data,
+    const size_t data_len,
+    const uint8_t* const signature,
+    const size_t signature_len,
+    const uint8_t* const pubkey,
+    const size_t pubkey_len) {
+  if (data_len == 0) {
     ERROR("Input length is zero.");
     return false;
   }
-  if (data == NULL){
+  if (data == NULL) {
     ERROR("Input data is null.");
     return false;
   }
-  if(signature == NULL){
+  if (signature_len != QRSIGNATURE_LEN) {
+    ERROR("%s: signature_len is: %zu, expected: %zu", __func__, signature_len, QRSIGNATURE_LEN);
```

Comment:
tbh we're not consistent about this. I personally like having the function names there, so if you don't mind I'll leave it.

It could potentially be rolled into the ERROR / INFO / DEBUG macros if we want it to be consistent everywhere ...